### ISSUE TITLE
[CUDA] mx.eval() fails when evaluating ConvTranspose2d gradients

### ISSUE THREAD

**Issue by ekamperi** [ml-explore/mlx#2516]
**Describe the bug**
`mx.eval()` fails with "[conv] Can not build op graph" when evaluating gradients from convolution operations with CUDA backend, making training impossible for models containing ConvTranspose2d layers. Forward pass and gradient computation work correctly on GPU. Everything works fine on CPU.

**To Reproduce**
```python
import mlx.core as mx
import mlx.nn as nn

# Simple NNet with ConvTranspose2d
class MiniNNet(nn.Module):
    def __init__(self):
        super().__init__()
        self.conv = nn.Conv2d(1, 8, 3, padding=1)
        self.up = nn.ConvTranspose2d(8, 1, 3, stride=2, padding=1, output_padding=1)

    def __call__(self, x):
        return self.up(self.conv(x))

# Force GPU usage
mx.set_default_device(mx.gpu)

model = MiniNNet()
x = mx.random.normal((2, 16, 16, 1))  # NHWC format

def loss_fn(model, x):
    return mx.mean(model(x) ** 2)

# This works
loss_and_grad_fn = nn.value_and_grad(model, loss_fn)
loss, grads = loss_and_grad_fn(model, x)

# This fails with "[conv] Can not build op graph"
mx.eval(grads)
```

**Expected behavior**
Gradient evaluation should work on GPU as it does on CPU.

**Desktop**
- MLX version: 0.28.0
- GPU: NVIDIA RTX 6000 Ada Generation
- CUDA: 12.9
- Driver version: 575.57.08
- Platform: Linux 5.15.0-151-generic x86_64

**Comment by awni** [comment]
I think this will be resolved by https://github.com/ml-explore/mlx/pull/2511, CC @zcbenz

**Comment by zcbenz** [comment]
I confirm https://github.com/ml-explore/mlx/pull/2511 has fixed this.
